packages/components/src/components/single-select/shadow.tsx

```diff
@@ -245,7 +245,7 @@ export class KolSingleSelect implements SingleSelectAPI {
 									}}
 									placeholder={this.state._placeholder}
 								/>
-								{this._inputValue && (
+								{this._inputValue && !this._hideClearButton && (
```

Comment:
Über den State gehen.
```suggestion
						{this._inputValue && !this.state._hideClearButton && (
```